core-bundle/src/Search/Indexer/DelegatingIndexer.php

```diff
@@ -30,22 +30,74 @@ public function addIndexer(IndexerInterface $indexer): self
 
     public function index(Document $document): void
     {
-        foreach ($this->indexers as $indexer) {
-            $indexer->index($document);
-        }
+        $this->wrapInTryCatch(
+            static function (IndexerInterface $indexer) use ($document): void {
+                $indexer->index($document);
+            },
+        );
     }
 
     public function delete(Document $document): void
     {
-        foreach ($this->indexers as $indexer) {
-            $indexer->delete($document);
-        }
+        $this->wrapInTryCatch(
+            static function (IndexerInterface $indexer) use ($document): void {
+                $indexer->delete($document);
+            },
+        );
     }
 
     public function clear(): void
     {
+        $this->wrapInTryCatch(
+            static function (IndexerInterface $indexer): void {
+                $indexer->clear();
+            },
+        );
+    }
+
+    private function wrapInTryCatch(callable $function): void
+    {
+        $warningsOnly = true;
+        $indexerExceptions = [];
+        $generalExceptions = [];
+
         foreach ($this->indexers as $indexer) {
-            $indexer->clear();
+            try {
+                $function($indexer);
+            } catch (IndexerException $exception) {
+                $indexerExceptions[] = $exception;
+                if (!$exception->isOnlyWarning()) {
+                    $warningsOnly = false;
+                }
+            } catch (\Throwable $exception) {
+                $generalExceptions[] = $exception;
+            }
         }
+
+        if ([] !== $generalExceptions) {
+            throw new \LogicException($this->getMergedExceptionMessage($generalExceptions));
```

Comment:
I think we should just throw the first of the collected exceptions (just like we do in `Cron`), otherwise you will not have the original exception - and thus no information about its exception chain and their stack traces.